src/main/kotlin/com/hibob/academy/employee_feedback_feature/service/ResponseService.kt

```
@@ -11,10 +11,14 @@ class ResponseService @Autowired constructor(
     private val responseDao: FeedbackResponseDao,
     private val feedbackDao: FeedbackDao
 ) {
-    fun submitResponse(newFeedbackResponse: ResponseSubmission, companyId: Long): Long {
+    fun submitResponse(newFeedbackResponse: ResponseSubmission, employeeId: Long, companyId: Long): Long {
         val feedbackExists = feedbackDao.getFeedbackByFeedbackId(companyId, newFeedbackResponse.feedbackId)
         feedbackExists?.let {
-            return responseDao.submitResponse(newFeedbackResponse)
+            if (!feedbackExists.isAnonymous) {
+                return responseDao.submitResponse(newFeedbackResponse, employeeId)
+            }
+
+            throw IllegalArgumentException("Sorry, this feedback is anonymous. You cannot respond to it!")
         }
 
         throw IllegalArgumentException("Feedback with ID ${newFeedbackResponse.feedbackId} does not exist.")
```

in this case, maybe throw not found. but illegal argument is also ok